### PR TITLE
(Review second) Update FI profile - phase3 (part II)

### PR DESCRIPTION
**Pull request by meissadia**
Part 2 of #222 

## Changes
- Populates form's `defaultValues` with existing Institution data
- Implements `Clear form` functionality
- Removes previous method of providing form input defaults
- Removes vague `id=stuff`
- Temp fix for TextArea resizing issue

## TODO
- Lots of Typescript stuff (Will work to address these issues in my next PR)

## How to test this PR
1. In browser console, ensure routing is enabled (`setIsRoutingEnabled(true)`)
1. Login with a user who has an associated institution
2. Click on the institution to `View` it's profile
3. Click on `Update your financial institution profile`
5. Modify a single field
6. Hit the `Clear form` button, observe that the edited data is reset
5. Modify a single field
8. Hit `Submit`
9. Verify you are redirected to the `/summary` route and shown a simulated "Success" confirmation
10. In console, look for `data to be submitted (before format)` and confirm the field you edited reflects the data you entered

### CODE REVIEW

File: src/pages/Filing/UpdateFinancialProfile/buildUfpDefaults.tsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+import type { InstitutionDetailsApiType } from '../ViewInstitutionProfile/institutionDetails.type';
+import type { UFPSchema } from './types';
+import { sblInstitutionTypeMap } from './types';
+
+const defaultsId = new Set(['primary_federal_regulator']);
+
+const defaultsCode = new Set(['hq_address_state']);
+
+const defaultsPlain = new Set([
+  'hq_address_street_1',
```
[the remark this answers is not in the document]
Comment (meissadia) [reply src/pages/Filing/UpdateFinancialProfile/buildUfpDefaults.tsx:11]: As discussed, resolving all TS issues will be a part of implementing form validations.

File: src/pages/Filing/UpdateFinancialProfile/index.tsx [hunk 2]
```diff
@@ -6,6 +6,7 @@ import type { JSXElement } from 'design-system-react/dist/types/jsxElement';
 import { useError500 } from 'pages/Error/Error500';
 import { useParams } from 'react-router-dom';
 import UFPForm from './UfpForm';
+import './updateFinancialProfile.less';
```
[the remark this answers is not in the document]
Comment (meissadia) [reply src/pages/Filing/UpdateFinancialProfile/index.tsx:9]: The CSS implemented in that file is a temp workaround for issues that should be addressed in the DSR, so we'll delete this once those adjustments are implemented.

File: src/pages/Filing/UpdateFinancialProfile/updateFinancialProfile.less [hunk 3]
```diff
@@ -0,0 +1,4 @@
+// TODO: Fix this in DSR: https://github.com/cfpb/design-system-react/issues/315
+#additional_details {
+  resize: vertical;
```
[the remark this answers is not in the document]
Comment (meissadia) [reply src/pages/Filing/UpdateFinancialProfile/updateFinancialProfile.less:3]: Added a consideration in for `max-height`: https://github.com/cfpb/design-system-react/issues/315

File: src/components/LabelOptional.tsx [hunk 4]
```diff
@@ -0,0 +1,11 @@
+function LabelOptional(): JSX.Element {
+  return (
+    <span className='text-base font-normal text-[#43484e]'> (optional)</span>
+  );
+}
+
+LabelOptional.defaultProps = {
+  isOptional: true,
```
[the remark this answers is not in the document]
Comment (meissadia) [reply src/components/LabelOptional.tsx]: Removed

File: src/pages/Filing/UpdateFinancialProfile/UpdateIdentifyingInformation.tsx [hunk 5]
```diff
@@ -72,7 +71,7 @@ function UpdateIdentifyingInformation({
   control: any;
   formErrors: string[];
 }): JSXElement {
-  const typeOtherData = data.sbl_institution_types?.find(item => {
+  const typeOtherData = data.sbl_institution_types.find(item => {
```
[the remark this answers is not in the document]
Comment (meissadia) [reply src/pages/Filing/UpdateFinancialProfile/UpdateIdentifyingInformation.tsx:74]: Updated to remove `string` checking.

[the remark this answers is not in the document]
Comment (meissadia) [reply src/pages/Filing/UpdateFinancialProfile/UpdateIdentifyingInformation.tsx:74]: I should also note that there's a lot of cleanup and simplification that is handled in the [next PR](https://github.com/cfpb/sbl-frontend/pull/283/files), so I don't want to get too hung up on details here.  Nor do I want to make to many additional changes that would then create conflicts with that PR.

I've added a [note on that PR](https://github.com/cfpb/sbl-frontend/pull/283#issuecomment-1992475802) to track any additional issues or items to focus on.

**Comment by billhimmelsbach** [comment]
@shindigira Can we get a re-review on this?

**Pull request merged**